src/main/java/com/google/edith/services/ReceiptFileHandlerService.java

```diff
@@ -0,0 +1,58 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.edith.services;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.FileInfo;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that stores file uploaded in Blobstore and serves that blob on succession. */
+public class ReceiptFileHandlerService {
+  private final BlobstoreService blobstoreService;
+
+  public ReceiptFileHandlerService(BlobstoreService blobstoreService) {
+    this.blobstoreService = blobstoreService;
+  }
+
+  /** Redirects the page to a new tab and serves the blob. */
+  public void serveBlob(HttpServletResponse response, List<FileInfo> fileKeys) throws IOException {
+    BlobKey fileBlobKey = getBlobKey(fileKeys);
+    blobstoreService.serve(fileBlobKey, response);
+  }
+
+  /**
+   * Returns a List of BlobKey that points to the uploaded files in the HTML form or null if the
+   * user didn't upload a file.
+   */
+  public Optional<List<FileInfo>> getUploadedFileUrl(
```

Comment:
Do we need to return an Optional list here? Generally when returning Lists we can default to returning an empty list, this way it's easier for callers to interact with the method.

Reply:
Yeah sure. I changed it to return an empty list instead. I was getting the list from a map which could have returned null if the ```formInputElementName``` key was not in the map.